### DISCUSSION TITLE
superset-frontend can't run, JSX cannot be parsed

### DISCUSSION THREAD

**Discussion by piewaer** [apache/superset#15696] [Q&A / Help]
### Expected results

I want to run the superset-frontend only.

### Actual results
Get lots of errors

#### Screenshots
![image](https://user-images.githubusercontent.com/87263497/125627666-d5f72ce0-591f-44aa-bc77-9a745e58047c.png)

#### How to reproduce the bug
1. In the superset-frontend folder, npm ci
2. npm run dev-server
3. 98% of the process, get lots of errors, includes the type error of the data and the error not found by the module

### Environment
- superset version: `1.2`
- python version: `python --version`
- node.js version: `14.17.`
- npm version: `7.19.1`

**Comment by junlincc** [comment]
moving issue to Q&A

**Comment by navono** [comment]
I have the same error, any suggestion?

**Comment by villebro** [comment]
@navono can you share some more details about your setup? The frontend build system has had massive changes in the last few weeks, so knowing what version/SHA you're on will be important.

**Comment by navono** [comment]
After spending a few hours, finally it worked.

This is my environment:
```
OS: Win10 (21H2)
WSL2: 
  Ubuntu 20.04
  Python: 3.8.10
  Node: 16.13.1
VS Code (ESLint, prettier)
superset branch: 1.4
```

Basically I followed the instructions of `CONTRIBUTING.md`, all instructions are executed in `WSL`.

Flask server
1. lack `libmysqlclient-dev`
2. psycopg2-binary: `2.8.5` -> `2.8.6`

Frontend
1.  lot of `style error`
2. `superset_text` error in textUtils.ts 
3. `Variable @root-entry-name is undefined` error in `antd`

Workaround
1. disable `eslint` in `webpack.config.js`:
    ```
    // new ForkTsCheckerWebpackPlugin({
    //   eslint: {
    //     files: './src/**/*.{ts,tsx,js,jsx}',
    //     memoryLimit: 4096,
    //   },
    // }),
    ```
2. an ignored file is referenced in the source code, which seems a bit unreasonable. Add `superset_text.yml` in the root directory with content:
    ```
    DB_IMAGES:
	    postgresql: ""
	    bigquery: ""
	    snowflake: ""
    ```
4. check [this issue](https://github.com/ant-design/ant-design/issues/33007), upgrade to the latest version, `webpack.config.js` -> `less-loader`:
    ```
    options: {
          sourceMap: isDevMode,
          javascriptEnabled: true,
          modifyVars: {
            'root-entry-name': 'default',
          }
        },
    ```

hope it helps.
